go/ct/gen/transient_storage.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package gen
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"golang.org/x/exp/maps"
+	"pgregory.net/rand"
+
+	"github.com/Fantom-foundation/Tosca/go/ct/common"
+	"github.com/Fantom-foundation/Tosca/go/ct/st"
+)
+
+type TransientStorageGenerator struct {
+	nonZeroConstraints map[Variable]bool
+	unsatisfiable      bool
+}
+
+func NewTransientStorageGenerator() *TransientStorageGenerator {
+	return &TransientStorageGenerator{nonZeroConstraints: map[Variable]bool{}}
+}
+
+// BindToNonZero binds a location in transient storage to a non-zero value.
+func (t *TransientStorageGenerator) BindToNonZero(key Variable) {
```

Comment:
nit: if generator is already unsatisfiable, return?

Reply:
The performance impact will be neglectable since this function will not be called a big number of times.